packages/ra-core/src/dataProvider/useInfiniteGetList.ts

```diff
@@ -80,7 +79,7 @@ export const useInfiniteGetList = <RecordType extends RaRecord = any>(
         Error,
         GetInfiniteListResult<RecordType>
     >(
-        [resource, 'getList', { pagination, sort, filter, meta }],
+        [resource, 'getInfiniteList', { pagination, sort, filter, meta }],
```

Comment:
This is necessary because otherwise the normal and infinite list queries share the same cache, which leads to errors because they don't have the same structure. 

This means that all the dataProvider hooks that optimistically update the `getList` query cache must also do it for the `getInfiniteList`.